lookout/style/typos/preprocessing/metrics.py

```diff
@@ -0,0 +1,194 @@
+                 has columns "typo", "corrupted".
+    :param suggestions: {id : [(candidate, correct_prob)]},
+                        candidates inside one suggestions list are
+                        sorted by correct_prob in a descending order.
+    :param k: Number of the first suggested corrections to check.
+    :return: Scores of suggestions for score@k problem.
+    """
+    scores = Scores(0, 0, 0, 0)
+    for i in data.index:
+        if data.loc[i, COLUMNS["TOKEN"]] != data.loc[i, COLUMNS["CORRECT_TOKEN"]]:
+            if data.loc[i, COLUMNS["CORRECT_TOKEN"]] in first_k_set(suggestions[i], k):
+                scores.tp += 1
+            else:
+                scores.fn += 1
+        else:
+            if data.loc[i, COLUMNS["CORRECT_TOKEN"]] in first_k_set(suggestions[i], k):
+                scores.tn += 1
+            else:
+                scores.fp += 1
+    return scores
+
+
+def score_at_k_on_corrections(data: pandas.DataFrame, suggestions: dict, k: int = 1) -> Scores:
+    """
+    Calculate score@k of solution for typo correction problem only on correcting suggestions.
+
+    The suggestions for typo correction are considered right if there is a right one among
+    the first k. Only the tokens, considered to be typoed, are taken into account
+    (correcting suggestions).
+    :param data: DataFrame which is indexed by "id" and has columns "typo", "corrupted".
+    :param suggestions: {id : [(candidate, correct_prob)]},
+                        candidates inside one suggestions list are
+                        sorted by correct_prob in a descending order.
+    :param k: Number of the first suggested corrections to check.
+    :return: Scores of correcting suggestions for score@k problem .
+    """
+    scores = Scores(0, 0, 0, 0)
+    for i in data.index:
+        if suggestions[i][0][0] != data.loc[i, COLUMNS["TOKEN"]]:
+            if data.loc[i, COLUMNS["TOKEN"]] != data.loc[i, COLUMNS["CORRECT_TOKEN"]]:
+                if data.loc[i, COLUMNS["CORRECT_TOKEN"]] in first_k_set(suggestions[i], k):
+                    scores.tp += 1
+                else:
+                    scores.fn += 1
+            else:
+                if data.loc[i, COLUMNS["CORRECT_TOKEN"]] in first_k_set(suggestions[i], k):
+                    scores.tn += 1
+                else:
+                    scores.fp += 1
+    return scores
+
+
+def accuracy(score: Scores) -> float:
+    """Calculate accuracy."""
+    return (score.tp + score.tn) / sum(list(score))
+
+
+def precision(score: Scores) -> float:
+    """Calculate precision."""
+    return score.tp / (score.tp + score.fp)
+
+
+def recall(score: Scores) -> float:
+    """Calculate recall."""
+    return score.tp / (score.tp + score.fn)
+
+
+def f1(score: Scores) -> float:
+    """Calculate f1 score."""
+    return 2 / (1 / precision(score) + 1 / recall(score))
+
+
+def print_score_metrics(score: Scores, file=None):
+    """Print score metrics."""
+    print(score, file=file)
```

Comment:
I see that this is a sort of report. Please change ti to Jinja2 templates (refer to https://github.com/src-d/style-analyzer/blob/master/lookout/style/format/benchmarks/general_report.py#L69)